index.js

```diff
@@ -95,13 +99,19 @@ module.exports = fp(function from (fastify, opts, next) {
       headers['content-length'] = 0
     }
 
-    this.request.log.info({ source }, 'fetching from remote server')
-
     const requestHeaders = rewriteRequestHeaders(req, headers)
 
-    request({ method: req.method, url, qs, headers: requestHeaders, body }, (err, res) => {
+    const requestConfig = { method: req.method, url, qs, headers: requestHeaders, body }
+
+    if (logProxyRequest) {
+      logProxyRequest.apply(this, [source, requestConfig])
```

Comment:
The docs should indicate that the function is invoked with `this` bound to the Fastify reply object. And that `this.log` should be used to actually log things within the function.